src/plugins/visualizations/public/plugin.ts

```diff
@@ -131,9 +140,11 @@ export class VisualizationsPlugin
     expressions.registerRenderer(visualizationRenderer);
     expressions.registerFunction(rangeExpressionFunction);
     expressions.registerFunction(visDimensionExpressionFunction);
-
-    const embeddableFactory = new VisualizeEmbeddableFactory({ start });
-    embeddable.registerEmbeddableFactory(VISUALIZE_EMBEDDABLE_TYPE, embeddableFactory);
+    this.embeddableFactory = new VisualizeEmbeddableFactory(
+      { start },
+      async () => (await core.getStartServices())[0].application
```

Comment:
You should already have Application service in that `start` variable above.

```ts
start().core.application
```

```suggestion
```

Reply:
Implemented this suggestion. It's a lot cleaner now, thanks @streamich !